clang/lib/CIR/CodeGen/CIRGenTypes.cpp

```diff
@@ -320,6 +321,53 @@ mlir::Type CIRGenTypes::convertType(QualType type) {
           cir::IntType::get(&getMLIRContext(), astContext.getTypeSize(ty),
                             /*isSigned=*/true);
       break;
+
+    // SVE types
+    case BuiltinType::SveInt8:
+      resultType =
+          cir::VectorType::get(builder.getSInt8Ty(), 16, /*isScalable=*/true);
+      break;
+    case BuiltinType::SveUint8:
+      resultType =
+          cir::VectorType::get(builder.getUInt8Ty(), 16, /*isScalable=*/true);
+      break;
+    case BuiltinType::SveInt16:
+      resultType =
+          cir::VectorType::get(builder.getSInt16Ty(), 8, /*isScalable=*/true);
+      break;
+    case BuiltinType::SveUint16:
+      resultType =
+          cir::VectorType::get(builder.getUInt16Ty(), 8, /*isScalable=*/true);
+      break;
+    // TODO: Waiting for FP type helpers
+    // case BuiltinType::SveFloat16:
+    //   resultType = cir::VectorType::get(builder.getF16Type(), 8,
```

Comment:
Can you add the type helpers? We already have `fP16Ty`, `floatTy`, and `doubleTy` in `CIRGenTypeCache` so it should be trivial.

Reply:
Addressed in this [commit](https://github.com/llvm/llvm-project/pull/172346/commits/51064ee89a887d8d498c2deca1a80e4a7974869c) - hopefully that's what you had in mind.